azavea/pfb-network-connectivity
# Water removal and residential speed defaults

## Overview

The changes include added functionality to:
1) Remove census blocks that contain no land area, which will improve the BNA maps by only visualizing blocks that have a BNA score. Block removal is based on a csv listing all 2010 Decennial census blocks that have zero land area.
2) Set default speed limits for residential streets based on the state, to account for states where the legislated residential speed limit is greater than 25 mph.
3) Set default speed limits for residential streets based on the city (takes precedence over state), to account for cities where the legislated residential speed limit is different than the BNA default of 25 mph or different than the state default (e.g. Georgia has a 30 mph residential limit, but the city of Dunwoody, GA has a 25 mph residential limit).

### Demo

Original Dunwoody, GA heat map:

![image](https://user-images.githubusercontent.com/13860074/46556503-19551c00-c8a4-11e8-8e0e-0df1c13b3ae0.png)

Dunwoody, GA heat map with water removed and state default residential speed implemented (30 mph):

![image](https://user-images.githubusercontent.com/13860074/46556516-25d97480-c8a4-11e8-8b94-3ff31fae6951.png)

Dunwoody, GA heat map with water removed, state default, and city default residential speed implemented (25 mph):

![image](https://user-images.githubusercontent.com/13860074/46556551-3853ae00-c8a4-11e8-851e-30ed94993eb4.png)

### Notes

* All external files (three csv files) are stored as public files on S3. 
* The city defaults require an additional input term, the city FIPS code, as an identifier.
* As of 10/5, running the analysis locally is failing due to https://github.com/azavea/ansible-pip/issues/9

## Testing Instructions

* Run local analysis for Dunwoody, GA with PFB_CITY_FIPS = 1324768
* Run local analysis for Dunwoody, GA without PFB_CITY_FIPS
* Compare results to https://bna.peopleforbikes.org/#/places/d8dfa19c-8048-46ad-ab01-80760fb21b27/
* See images of expected results above
